src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/Example2.java

```diff
@@ -0,0 +1,28 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="RequireThis">
+      <property name="checkMethods" value="false"/>
+    </module>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+// xdoc section -- start
+class Example2 {
+  int a,b,c;
+
+  Example2(int a) {
+    // overlapping by constructor argument
+    this.a = a; // OK, this keyword used
+    b = 0; // OK, no overlap
+    foo(5); // OK, no validation for methods
+  }
+
+  void foo(int c) {
+    // overlapping by method argument
+    c = c; // violation, reference to instance variable "c" requires "this"
```

Comment:
After all above changes you should spot yourself that example1 and example2 are the same for violation. So we changed config, but behavior didn't change.
Can we update examples to make violation to popup or be removed each time we change configuration.

All examples should be different by violations.

Reply:
I am getting a lil confused. Technically there should not be any violation in line 25 in Example2 since   <property name="checkMethods" value="false"/> but still it is getting flagged for violation why is that?

Reply:
looks like we run into https://github.com/checkstyle/checkstyle/issues/4216

Reply:
please proceed with renaming that requested in other review items.

Please try to read logic of RequireThisCheck.java to understand when this property make a difference on behavior, but please do this after all above review items.

Reply:
we need this violation https://github.com/checkstyle/checkstyle/blob/8774a4018cd69344f8e4e985834e4de91cbc1691/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisEnumInnerClassesAndBugs.java#L26 to appear and disappear in Examples.

Reply:
@AmitKumarDeoghoria , if you stuck, just fix what you can ,push to GitHub, I will assist.

Reply:
I tried implementing taking reference from whats's given in the "InputRequireThis.." but it's still not getting flagged. I have implemented the above reviews. Let me know what should be the next steps.

Reply:
I am AFK, so need help to run CLI or just extend example to see it this works.

We might have defect that or some nuance that method in ctor are not validated.
Please add method to example:
```
void method2(int i) {
        foo(i); // violation 'Method call to 'method1' needs "this.".'
}
```

Reply:
I got it, `validateOnlyOverlapping = false` is required to raise violation. So we can not have violation on method here.

Reply:
Done.